collectors/modality-probe-debug-collector/src/lib.rs

```diff
@@ -0,0 +1,1169 @@
+    pub fn collect_report(&mut self) -> Result<Report, DebugCollectorError> {
+        let num_missed = self.reader.read(&mut self.rbuf)?;
+        // Possibly add entries missed event
+        if num_missed > 0 {
+            let num_missed_rounded = u64::min(num_missed, u32::MAX as u64) as u32;
+            let (ev, payload) =
+                LogEntry::event_with_payload(EventId::EVENT_LOG_ITEMS_MISSED, num_missed_rounded);
+            self.rbuf.insert(0, WholeEntry::Double(ev, payload));
+        }
+
+        // Keep a copy of the clocks before this report which will be used as frontier clocks
+        let report_clocks = self.clocks.clone();
+
+        // Merge clocks from this report
+        for entry in self.rbuf.iter() {
+            if let WholeEntry::Double(first, second) = entry {
+                if first.has_clock_bit_set() {
+                    let id = ProbeId::new(first.interpret_as_logical_clock_probe_id())
+                        .ok_or_else(|| DebugCollectorError::LogProcessingError)?;
+                    let (epoch, ticks) = modality_probe::unpack_clock_word(second.raw());
+                    Self::merge_clock(&mut self.clocks, LogicalClock { id, epoch, ticks })
+                }
+            }
+        }
+
+        let report =
+            Report::try_from_log(self.clocks[0], self.seq_num, report_clocks, &self.rbuf[..])
+                .map(|report| {
+                    self.seq_num += 1;
+                    report
+                })
+                .map_err(|_e| DebugCollectorError::ReportSerializationError);
+        self.rbuf.clear();
+        report
+    }
+
+    fn merge_clock(clocks: &mut Vec<LogicalClock>, ext_clock: LogicalClock) {
+        let mut existed = false;
+        for c in clocks.iter_mut() {
+            if c.id == ext_clock.id {
+                if OrdClock(ext_clock.epoch, ext_clock.ticks) > OrdClock(c.epoch, c.ticks) {
+                    c.epoch = ext_clock.epoch;
+                    c.ticks = ext_clock.ticks;
+                }
+                existed = true;
+            }
+        }
+        if !existed {
+            clocks.push(ext_clock);
+        }
+    }
+
+    /// Write to "write priority" field in probe
+    pub fn set_overwrite_priority(&mut self, level: u32) -> Result<(), DebugCollectorError> {
+        self.priority_writer.write(level)
+    }
+}
+
+/// Open memory accessor based on config
+fn open_mem_accessor(c: &Config) -> Result<Rc<RefCell<dyn MemoryAccessor>>, DebugCollectorError> {
+    match &c.target {
+        Target::ProbeRsTarget(target) => {
+            let probes = probe_rs::Probe::list_all();
+            if probes.is_empty() {
+                return Err(DebugCollectorError::TargetError);
+            }
+            let probe = probes[0]
+                .open()
+                .map_err(|_e| DebugCollectorError::TargetError)?;
+            let session = probe
+                .attach(target)
+                .map_err(|_e| DebugCollectorError::TargetError)?;
```

Comment:
The lack of detailed information in this set of `TargetError` instances (compared to the other ways a `DebugCollectorError::TargetError` might get tossed in this system) would make remediation of the error tricky, I imagine. Is there some further DebugCollectorError variants or sub-errors nested within a DebugCollectorError variant that could be developed to make a correction path apparent to the end user?